test.lua

```diff
@@ -3783,6 +3783,17 @@ function nntest.View()
       "Error in minibatch generalization nElement")
 end
 
+function nntest.ViewNonContiguous()
+  -- Non-continous input and gradOutput
+  local nonContigInput = torch.randn(2, 2, 2):select(2, 1)
+  local nonContigGradOut = torch.randn(4, 2):select(2, 2)
+  local viewModule = nn.View(4)
+  mytester:assert(nonContigInput:isContiguous() == false)
```

Comment:
I think those asserts should be changed.
What I think you could do is to create a `contigInput`, which is a clone of `nonContigInput`, run both in the View module, and assert that the output tensors are equal.
Same thing for gradInput.